.github/workflows/maven-publish.yml

```diff
@@ -104,6 +109,11 @@ jobs:
           fetch-tags: ${{ inputs.version_strategy == 'tag' && 'true' || 'false' }}
           fetch-depth: 0
 
+      - name: Copy maven settings
+        if: ${{ inputs.settings-xml_file != '' }}
+        run: |
+          wget ${{ inputs.settings-xml_file }} -O ~/.m2/settings.xml
+
       - name: Set up Java ${{ inputs.java_version }}
```

Comment:
This step should be before Copy maven setting, otherwise it will override custom settings.xml?